packages/edit-site/src/components/sidebar/global-styles-sidebar.js

```diff
@@ -186,63 +250,73 @@ export default function GlobalStylesSidebar() {
 				</>
 			}
 		>
-			<Navigation>
-				<NavigationMenu>
-					<GlobalStylesLevel
-						context={ root }
-						getStyle={ getStyle }
-						setStyle={ setStyle }
-						getSetting={ getSetting }
-						setSetting={ setSetting }
-					/>
-					<NavigationGroup className="edit-site-global-styles-sidebar__blocks-group">
-						<NavigationItem
-							className="edit-site-global-styles-sidebar__blocks-group-help"
-							isText
-						>
-							{ __(
-								'Customize the appearance of specific blocks for the whole site'
-							) }
-						</NavigationItem>
-						<NavigationItem
-							item="item-blocks"
-							navigateToMenu="blocks"
-							title={ __( 'Blocks' ) }
-						/>
-					</NavigationGroup>
-				</NavigationMenu>
-				<NavigationMenu
-					menu="blocks"
-					parentMenu="root"
-					title={ __( 'Blocks' ) }
-				>
+			<Navigator initialPath="/">
+				<NavigatorScreen path="/" exact>
+					<GlobalStylesLevelMenu context={ root } />
+
+					<ItemGroup>
+						<Item>
+							<p>
+								{ __(
+									'Customize the appearance of specific blocks for the whole site'
+								) }
+							</p>
+						</Item>
+						<NavigationButton path="/blocks">
+							{ __( 'Blocks' ) }
+						</NavigationButton>
+					</ItemGroup>
+				</NavigatorScreen>
+
+				<NavigatorScreen path="/blocks" exact>
+					<ScreenHeader back="/" title={ __( 'Blocks' ) } />
 					{ map( blocks, ( _, name ) => (
-						<NavigationItem
+						<NavigationButton
+							path={ '/blocks/' + name }
 							key={ 'menu-itemblock-' + name }
-							item={ 'block-' + name }
-							navigateToMenu={ 'block-' + name }
-							title={ getPanelTitle( name ) }
-						/>
+						>
+							{ getPanelTitle( name ) }
+						</NavigationButton>
 					) ) }
-				</NavigationMenu>
+				</NavigatorScreen>
+
 				{ map( blocks, ( block, name ) => (
```

Comment:
For blocks that don't support any global styles panel, we should not render a menu for them. I think we regressed on https://github.com/WordPress/gutenberg/pull/34885.

Reply:
We can leave this separate I think to avoid having a diff unrelated with the change of the navigation components.